### PR TITLE
[FEATURE] [MER-3303] Style project view navigation

### PR DESCRIPTION
**Pull request by simonchoxx**
[MER-3303](https://eliterate.atlassian.net/browse/MER-3303)

This PR implements the navigation of a project in the sidebar for an author's workspace.

It adds the menus and sub-menus that already exist in the navigation of a project. 

The routes are modified, generating new routes that will be implemented and that are under the scope of the author's workspace.

For the time being, as discussed with the development team, when accessing a certain project, you will be redirected to the overview of the selected project, but navigation will be maintained with the current layout. 

- Navigate with current layout

https://github.com/user-attachments/assets/d2934d4e-ee20-4cbe-82c9-09e7995eac48

- New project sidebar

https://github.com/user-attachments/assets/ae69f490-feac-4e79-b294-94a9988b2559

[MER-3303]: https://eliterate.atlassian.net/browse/MER-3303?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/oli_web/live/workspaces/course_author/overview_table_model.ex [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  alias OliWeb.Common.SessionContext
+  alias OliWeb.Common.Table.{ColumnSpec, SortableTableModel}
+  alias OliWeb.Router.Helpers, as: Routes
+
+  def new(%SessionContext{} = ctx, sections) do
+    column_specs = [
+      %ColumnSpec{
+        name: :title,
+        label: "Title",
+        render_fn: &__MODULE__.custom_render/3
```
Comment (manelli) [inline lib/oli_web/live/workspaces/course_author/overview_table_model.ex:13]: Nitpick: Using `__MODULE__` is not necessary,  `render_fn: &custom_render/3` should work

**Review by manelli — APPROVED** [review]
Awesome work! 🚀 
Left some minor comments

File: lib/oli_web/components/delivery/layouts.ex [hunk 2]
```diff
@@ -471,15 +500,36 @@ defmodule OliWeb.Components.Delivery.Layouts do
...
-    case active_workspace do
-      :course_author -> ~p"/workspaces/course_author?#{params}"
-      :instructor -> ~p"/workspaces/instructor?#{params}"
-      :student -> ~p"/workspaces/student?#{params}"
+    base_module =
+      case active_workspace do
+        :course_author -> OliWeb.Workspaces.CourseAuthor
+        :instructor -> OliWeb.Workspaces.Instructor
+        :student -> OliWeb.Workspaces.Student
+        _ -> raise {"Unknown workspace: #{active_workspace}"}
```
[the remark this answers is not in the document]
Comment (Francisco-Castro) [reply lib/oli_web/components/delivery/layouts.ex:513]: @manelli It looks like `raise/2` is the one that expects a module and a message. The arity one (`raise/1`) accepts just a string.

Ref: https://hexdocs.pm/elixir/Kernel.html#raise/1

[the remark this answers is not in the document]
Comment (manelli) [reply lib/oli_web/components/delivery/layouts.ex:513]: I meant, name or string or exception. I think with a tuple will fail

[the remark this answers is not in the document]
Comment (manelli) [reply lib/oli_web/components/delivery/layouts.ex:513]:
```elixir
raise {"hello"}
# => ** (ArgumentError) raise/1 and reraise/2 expect a module name, string or exception as the first argument, got: {"hello"}
    iex:7: (file)
 ```

[the remark this answers is not in the document]
Comment (Francisco-Castro) [reply lib/oli_web/components/delivery/layouts.ex:513]: I think that's what @simonchoxx is looking for ---to throw an error. However, you make a good point; perhaps a warning log would be more appropriate.

[the remark this answers is not in the document]
Comment (manelli) [reply lib/oli_web/components/delivery/layouts.ex:513]: but its the wrong error 😆 
the correct line would be: raise `"Unknown workspace: #{active_workspace}"` (minus the `{` and `}`)

[the remark this answers is not in the document]
Comment (Francisco-Castro) [reply lib/oli_web/components/delivery/layouts.ex:513]: How did I miss the curly braces? 🫤

[the remark this answers is not in the document]
Comment (simonchoxx) [reply lib/oli_web/components/delivery/layouts.ex:513]: Nice catch @manelli! You are right!

**Review by eliknebel — APPROVED** [review]

**Pull request merged**